src/dev/flang/fe/Module.java

```diff
@@ -269,6 +268,62 @@ else if (existing == f && !existing.generics().equals(f.generics()) ||
   }
 
 
+  /**
+   * Is type defined by feature `af` visible in file `usedIn`?
+   * If `af` does not define a type, result is false.
+   *
+   * @param usedIn
+   * @param af
+   * @return
+   */
+  protected boolean typeVisible(SourceFile usedIn, AbstractFeature af)
+  {
+    var m = (af instanceof LibraryFeature lf) ? lf._libModule : this;
+    var definedIn = af.pos()._sourceFile;
+    var v = af.visibility();
+
+    // NYI check if generics/constraints visible?
```

Comment:
This should not be checked here, but at the declaration of `af`.

Reply:
I removed the comment.